src/Microsoft.OData.Core/UrlValidation/ValidationEngine/ODataUrlValidator.cs

```diff
@@ -411,26 +438,29 @@ private void ValidateImpliedProperties(IEdmType segmentType, SelectExpandClause
         private void ValidateProperties(IEdmType edmType, ODataUrlValidationContext context)

```

Comment:
Could this be called multiple times for same edmtype? if so can this be cahced

Reply:
there's not really anything to cache (the method looks for and executes any rules that are applied to the particular type, and the rules add any exceptions to the validation context). However, we go one better, and on line 441 we skip validation if we've already run validation on this type. 

---
In reply to: [518476690](https://github.com/OData/odata.net/pull/1917#discussion_r518476690) [](ancestors = 518476690)